guides/data_modelling/contexts.md

```diff
@@ -4,11 +4,39 @@ Phoenix guides are broken into several major sections. The main building blocks
 
 When building a Phoenix project, we are first and foremost building an Elixir application. Phoenix's job is to provide a web interface into our Elixir application. Naturally, we compose our applications with modules and functions, but we often assign specific responsibilities to certain modules and give them names: such as controllers, routers, and live views.
 
-However, the most important part of your web application is often where we encapsulate data access and data validation. We call these modules **contexts**. They often talk to a database, using `Ecto`, or APIs, using an HTTP client such as `Req`. By giving modules that expose and group related data the name **contexts**, we help developers identify these patterns and talk about them. At the end of the day, contexts are just modules, as are your controllers, views, etc.
+However, the most important part of your web application is often where we encapsulate data access and data validation. We call these modules **contexts**. They often talk to a database, using `Ecto`, or APIs, using an HTTP client such as `Req`. By giving these modules a name, we help developers identify these patterns and talk about them. At the end of the day, contexts are just modules, as are your controllers, views, etc.
 
-Contexts are not a new concept either. For example, anytime you call Elixir's standard library, be it `Logger.info/1` or `Stream.map/2`, they are the entrypoint to rich functionality, often implemented across multiple modules which are hidden from us. **Contexts** in Phoenix are the same: plain modules that act as boundaries to decouple and isolate parts of your application.
+If you have used `mix phx.gen.html`, `mix phx.gen.json`, or `mix phx.gen.live`, you have already used contexts. For example, run the following generator in a Phoenix application:
 
-Let's use these ideas to build out our web application. Our goal is to build an ecommerce system where we can showcase products, allow users to add products to their cart, and complete their orders. Opposite to other Phoenix guides, **these guides are meant to be read in order**.
+```console
+$ mix phx.gen.live Post posts title body:text
+```
+
+The command above will output a few files, among them, a `MyApp.Posts.Post` schema in `lib/my_app/posts/post.ex`, which outlines how the resource is represented in the database, and a **context** module named `MyApp.Posts` that encapsulates all the database access to said schema. The `MyApp.Posts` module centralizes all functionality related to posts, instead of scattering logic around controllers, LiveViews, etc.
+
+Contexts are also useful to nest resources. For example, if you are adding comments to your posts, you can colocate their schemas, since comments belongs to posts, like this:
```

Comment:
```suggestion
Contexts are also useful to nest resources. For example, if you are adding comments to your posts, you can colocate their schemas, since comments belong to posts, like this:
```